Sources/Fluent/SQL/GeneralSQLSerializer.swift

```diff
@@ -469,17 +486,38 @@ open class GeneralSQLSerializer: SQLSerializer {
         clause += "JOIN"
         clause += sql(join.foreign.entity)
         clause += "ON"
-        clause += "\(sql(join.local.entity)).\(sql(join.local.idKey))"
+
+        let localKey: String
+        let foreignKey: String
+
+        switch join.child {
+        case .foreign:
+            localKey = join.local.idKey
+            foreignKey = join.local.foreignIdKey
```

Comment:
this is super confusing to me. `foreignKey` as `local.foreignIdKey` seems very odd, I think we need better naming conventions for this stuff